tests/test_time_series_mapper.py

```diff
@@ -0,0 +1,149 @@
+        dfd = pd.concat([dfd, dfd.assign(id=2).assign(value=dfd.value * 2)], axis=0)
+        return dfd
+
+    if one_week:
+        dfd = generate__one_week_per_month_by_hour()
+    else:
+        dfd = generate__one_weekday_day_and_one_weekend_day_per_month_by_hour()
+
+    # Add more data
+    df = pd.DataFrame(
+        {
+            "id": [1, 2],
+            "geography": ["IL", "CO"],
+            "time_zone": ["US/Central", "US/Mountain"],
+        }
+    )
+    dfd = dfd.merge(df, on="id", how="left")
+
+    return dfd
+
+
+def get_destination_schema(tzinfo) -> TableSchema:
+    schema = TableSchema(
+        name="mapped_data",
+        time_config=DatetimeRange(
+            start=datetime(year=2018, month=1, day=1, tzinfo=tzinfo),
+            resolution=timedelta(hours=1),
+            length=8760,
+            interval_type=TimeIntervalType.PERIOD_BEGINNING,
+            time_column="timestamp",
+        ),
+        time_array_id_columns=["id", "geography", "time_zone"],
+        value_column="value",
+    )
+    return schema
+
+
+def get_data_schema(one_week: bool = False) -> TableSchema:
+    if one_week:
+        time_format = RepresentativePeriodFormat.ONE_WEEK_PER_MONTH_BY_HOUR
+    else:
+        time_format = (
+            RepresentativePeriodFormat.ONE_WEEKDAY_DAY_AND_ONE_WEEKEND_DAY_PER_MONTH_BY_HOUR
+        )
+    schema = TableSchema(
+        name="load_data",
+        time_config=RepresentativePeriodTimeRange(
+            interval_type=TimeIntervalType.PERIOD_BEGINNING,
+            time_format=time_format,
+        ),
+        time_array_id_columns=["id", "geography", "time_zone"],
+        value_column="value",
+    )
+    return schema
+
+
+def get_timeseries(time_config: DatetimeRange) -> pd.Series:
+    ts = pd.date_range(
+        start=time_config.start, freq=time_config.resolution, periods=time_config.length
+    ).rename(time_config.time_column)
+    return ts
+
+
+def run_test(engine: Engine, one_week: bool = False, tzinfo: ZoneInfo | None = None) -> None:
+    engine.clear_compiled_cache()
+
+    # Generate
+    df = generate_data(one_week=one_week)
+
+    # Ingest
+    metadata = MetaData()
+    schema = get_data_schema(one_week=one_week)
+    with engine.connect() as conn:
+        write_database(df, conn, schema)
+        conn.commit()
+    metadata.reflect(engine, views=True)
+
+    ## Map
+    dest_schema = get_destination_schema(tzinfo=tzinfo)
+    map_time(engine, metadata, schema, dest_schema)  # this creates a table
+
+    # Check mapped table
+    with engine.connect() as conn:
+        query = f"select * from {dest_schema.name}"
+        queried = read_database(query, conn, dest_schema)
+    queried = queried.sort_values(by=["id", "timestamp"]).reset_index(drop=True)
+
+    truth = get_timeseries(dest_schema.time_config)
+    check_mapped_table(engine, queried, truth)
+
+
+def check_mapped_table(engine: Engine, dfs: pd.DataFrame, ts: pd.Series) -> None:
```

Comment:
Sorry if this is nit-picky, I'm just trying to understand conventions. Does `dfs` mean something?